appformer-kogito-bridge/src/main/java/org/appformer/kogito/bridge/client/context/EditorContextProvider.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.appformer.kogito.bridge.client.context;
+
+import java.util.Optional;
+import java.util.function.Function;
+
+import javax.enterprise.context.ApplicationScoped;
+
+import org.appformer.kogito.bridge.client.context.interop.EditorContextWrapper;
+
+/**
+ * 
+ * CDI Bean to provide access to EditorContext properties
+ *
+ */
+@ApplicationScoped
+public class EditorContextProvider {
+
+    /**
+     * The channel where the editor is running
+     * @return
+     */
+    public Optional<String> getChannel() {
+        return nullSafe(wrapper -> wrapper.getChannel());
+    }
+
+    private Optional<String> nullSafe(Function<EditorContextWrapper, String> action) {
+        EditorContextWrapper wrapper = EditorContextWrapper.get();
+        if (wrapper != null) {
+            return Optional.of(action.apply(wrapper));
+        }
+        return Optional.empty();
+    }
```

Comment:
I think it is more readable as

```java
 public Optional<String> getChannel() {
      return Optional.ofNullable(EditorContextWrapper.get()).map(EditorContextWrapper::getChannel);
 }
```

Reply:
> I think it is more readable as
> 
> ```java
>  public Optional<String> getChannel() {
>       return Optional.ofNullable(EditorContextWrapper.get()).map(EditorContextWrapper::getChannel);
>  }
> ```

@jesuino One question: what does Optional.empty would mean from a functional point of view? Is there a possible situation where the "channel" is empty ?

Reply:
Presumably, given the current implementation, the _channel_ would be empty when `IsKogito` is `true`. However I'm proposing `IsKogito` is killed-off and _channel_ **always** returns a value... but let's wait for @jesuino to advise.

Reply:
What I was testing yesterday was something  like:

If you are running Kogito and in a channel, so: isKogito==true and channel==channel where the editor is running
If you are running on Business Central or in a webapp: isKogito==false
When you test the editor in the browser (without any channel), channel.isEmpty() == true

Now adding BUSINESS-CENTRAL channel we simply would have an additional channel for Business Central only or channel.isEmpty ==true if no channel. This is what I am working right now.

Reply:
As long as we only have one place to check the environment in which an editor is running I'm happy... i.e. kill off `IsKogito` everywhere. My preference is to simply use `EditorContextProvider.getChannel()` to determine _where_ the editor is running, be it VSCode, Chrome, Business Central or elsewhere.

Reply:
@jesuino @manstis 
I think your comments may be merged in one solution (correct me if I'm wrong):
1) always return the channel identifier in one single place
2) do not rely on IsKogito (as you see, using it lead to a matrix of cases)

I personally agree, but still not really like the overall idea that somehow the content (i.e. the editor) should know about the container to work correctly - it seems against the concept of modularity - I have the impression there is some architectural issue behind that this "channel" information is meant to solve.
Beside that, `Optional.empty()` is a nice way to say `null`, and `null` has not any semantical meaning: I would prefer to always return a value or throw an exception.

Reply:
My understanding is that _channel_ may be used to determine where to show docks and which may be open by default etc based upon the container/channel... we could achieve this by composing different plugins/extensions for different containers/channels but IMO that could all too easily explode into a maintenance nightmare! There may be other uses of which I am unaware..

Reply:
@manstis 
If I follow correctly, that would lead to the following scenario:
1) editor must invoke the "channel" to know where they are running
2) basing on that, the editor itself must implement a sort of "if" - based on the channel, the editor should behave/look differently
3) this could go far away of the simple "screen container" approach I've understood from the beginning - it would require to spread the consequences of such "if" all over the code.
But what I still fail to understand is: we already have two specific modules - one is meant for kogito (and with it I mean client-side only - maybe this is my misunderstanding) and the other for business central. What other differentiation could possible be? *If* we are saying that some behavior of an editor inside VSCode must be different from the github extension, then we are saying that potentially for every new possible channel a different implementation should be provided, and that would lead to nightmare too, IMHO. 
I think that editors should be thought as composable pieces - with the wrapper layer provided only to allow the client-side requirements. Beside that, they should be absolutely agnostic about the container they are running in.

Reply:
Hi @gitgabrio IDK the full purpose of _channel_ other than what I've said above that _some_ editors in _some_ channels _may_ need to behave differently - and I don't mean offer different feature sets.. We have 2 wrappers. Plain and simple. One for (current) Business Central and one for kogito (be it VSCode, GitHub etc). The feature sets in each wrapper remain constant however I've heard mention of, for example (and I may be mis-quoting), the Properties Panel (in DMN and BPMN) may be required to be open by default in VSCode but not GitHub.

Reply:
If _channel_ is meant to be providing the meta-data for something else perhaps @jesuino or someone in foundation can explain it's exact purpose.

Reply:
@manstis @yesamer 
So, if I understand correctly, that _channel_ has some meaning only inside the client-side editors.
Having said that, IMO it looks much more like a "configuration" parameter, that should be set once and for all depending on the environment, and not something to be continuously invoked at runtime.  With that said, I think it would be better to move that during "compilation/build" (or whatever) phase: define a "channel" property that is set when the code is embedded in the container (VSCode, other) and then the editor behave differently based on that variable. I think the behavior would be much easier to predict and the code easier to test (not saying avoiding unpredictable method invocation depending on who-know-what mechanism)

Reply:
Hi everyone, 

EditorContextProvider.getChannel()  will return an Enum with

- DEFAULT (on any web app running based on AppFormer);
- On Kogito we will return VSCODE|GITHUB|Online based on the channel that the envelope is running;

Our webapps to ensure the correct values are returned in -testing and -runtime will have to implement an 
@Alternative for EditorContextProvider that will return the desired channel mock.

I also agree with Michael that is a good idea to remove isKogito() API after this PR gets merged.

Thanks everyone for the discussion. I hope that makes sense and provide value for us.

Reply:
updated PR with commit https://github.com/kiegroup/appformer/pull/873/commits/c6760d6b37ac09966d52feb72b92998a944c5652